test/test_helper.exs

```diff
@@ -44,47 +40,66 @@ defmodule FakeMarathon do
       :cowboy_req.stream_body(ev, :nofin, req)
       {:ok, req, state}
     end
+
+    def info(:close, req, state) do
```

Comment:
Is there any way to define a protocol for this kind of thing?

Reply:
This is a loop handler, described at https://ninenines.eu/docs/en/cowboy/2.1/guide/loop_handlers/ in the cowboy docs. The cowboy request handler process calls `init/2` (as with a normal handler) and then, if `init/2` returns a `{:cowboy_loop, ...}` tuple, waits for messages and passes them to `info/3` for further processing.

In Elixir, "protocol" has a specific meaning that doesn't seem to apply here. Did you means something else?

Reply:
I knew it had a specific meaning, but guess I meant in the general sense. I think I actually misunderstood and thought that anything that could listen for SSE events needed to have a `close` but this is just a `FakeMarathon` thing for tests.

Reply:
Ah. I'm not aware of any mechanism for specifying the set of messages a particular process will accept, but the standard way to do this sort of thing is with functions that send the appropriate messages. I'll add those to this handler module to make things more explicit.